ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -5839,13 +5842,33 @@ public HiveMetaHook getHook(
       }
     };
 
+    if ("rest".equals(conf.get("metastore.type", "hms"))) {
```

Comment:
1. conf.get(`catalog.type`)
2. can we move `rest` to constant or enum - CatalogType (HIVE, REST)

Reply:
Let's handle this in this ticket:
https://issues.apache.org/jira/browse/HIVE-28879